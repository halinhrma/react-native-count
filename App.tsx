/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */
import React, { useState } from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';

const App: React.FC = () => {
  const [count, setCount] = useState<number>(0);

  const handleIncrement = () => {
    setCount(count + 1);
  };

  return (
    <View style={styles.container}>
      <Text style={styles.countText}>Count: {count}</Text>
      <TouchableOpacity style={styles.button} onPress={handleIncrement} testID='incrementButton'>
        <Text style={styles.buttonText}>Increment</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  countText: {
    fontSize: 24,
    marginBottom: 20,
  },
  button: {
    backgroundColor: '#007BFF',
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderRadius: 5,
  },
  buttonText: {
    color: '#FFFFFF',
    fontSize: 18,
  },
});



export default App;
