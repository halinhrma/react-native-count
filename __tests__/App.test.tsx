/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

import renderer, { act } from 'react-test-renderer';
import { describe, test, expect } from "@jest/globals"
import { Text, TouchableOpacity } from 'react-native';

describe('App', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

    test('Clicking "Increment" button with 100 press', () => {
    const tree = renderer.create(<App />);
    const instance = tree.root;
    const button = instance.findByType(TouchableOpacity);
    for (let i = 1; i <= 100; i++) {
      act(() => {
        button.props.onPress();
      });
    }
    const countText = instance.findAllByType(Text)[0];
    const countTextContent = countText.props.children.join(''); 
    expect(countTextContent).toBe('Count: 100');
  });

  /**
   * Test normal
   */
  test('initial count is 0', () => {
    const tree = renderer.create(<App />);
    const instance = tree.root;
    const countText = instance.findAllByType(Text)[0];
    const countTextContent = countText.props.children.join(''); 
    expect(countTextContent).toBe('Count: 0');
  });

  test('button text is "Increment"', () => {
    const tree = renderer.create(<App />);
    const instance = tree.root;
    const buttonText = instance.findByType(TouchableOpacity).findByType(Text);
    expect(buttonText.props.children).toBe('Increment');
  });

  test('clicking "Increment" button increments count', () => {
    const tree = renderer.create(<App />);
    const instance = tree.root;
    const button = instance.findByProps({ testID: 'incrementButton' });
    act(() => {
      button.props.onPress();
    });
    const countText = instance.findAllByType(Text)[0];
    const countTextContent = countText.props.children.join(''); 
    expect(countTextContent).toBe('Count: 1');
  });


  test('button background color is "#007BFF"', () => {
    const tree = renderer.create(<App />);
    const instance = tree.root;
    const button = instance.findByProps({ testID: 'incrementButton' });
    expect(button.props.style.backgroundColor).toBe('#007BFF');
  });

  test('matches snapshot after multiple button presses', () => {
    const tree = renderer.create(<App />);
    const instance = tree.root;
    const button = instance.findByProps({ testID: 'incrementButton' });
    for (let i = 1; i <= 2; i++) {
      act(() => {
        button.props.onPress();
      });
    }
    const treeAfterPresses = tree.toJSON();
    expect(treeAfterPresses).toMatchSnapshot();
  });

  test("test borderRadius style",()=>{
    const testRenderer = renderer.create(<App />);
    const testInstance = testRenderer.root;
    expect(testInstance.findByType(TouchableOpacity).props.style.borderRadius).toBe(5);
  });

  test("test paddingVertical style",()=>{
    const testRenderer = renderer.create(<App />);
    const testInstance = testRenderer.root;
    expect(testInstance.findByType(TouchableOpacity).props.style.paddingVertical).toBe(10);
  });

  /**
   * End Test normal
   */



});
