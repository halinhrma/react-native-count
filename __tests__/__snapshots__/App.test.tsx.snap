// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot after multiple button presses 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "marginBottom": 20,
      }
    }
  >
    Count: 
    2
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#007BFF",
        "borderRadius": 5,
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
    testID="incrementButton"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 18,
        }
      }
    >
      Increment
    </Text>
  </View>
</View>
`;

exports[`App renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "marginBottom": 20,
      }
    }
  >
    Count: 
    0
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#007BFF",
        "borderRadius": 5,
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
    testID="incrementButton"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 18,
        }
      }
    >
      Increment
    </Text>
  </View>
</View>
`;
